.evergreen/config_generator/components/funcs/fetch_det.py

```diff
@@ -17,6 +17,17 @@ class FetchDET(Function):
             ''',
         ),
 
+        # Make shell scripts executable.
+        bash_exec(
+            command_type=EvgCommandType.SETUP,
+            working_dir="drivers-evergreen-tools",
+            script='''\
+                for file in $(find .evergreen -type f -name \*.sh); do
+                    chmod +x "$file" || exit
+                done
+            ''',
```

Comment:
```suggestion
            script='find .evergreen -type f -name "*.sh" -execdir chmod +x "{}" \;'
```

You can also include `+r` as Ezra suggested, but I'm not convinced that's necessary.

Reply:
> I'm not convinced it's necessary.

Absence of read permission bit me a few times during CDRIVER-3620 work. Apparently execute-only permission is (normally) only supported for binaries due to how they are executed; it does not work with interpreted shell scripts (see: https://unix.stackexchange.com/questions/34202/can-a-script-be-executable-but-not-readable).

Reply:
I'm confused about how it would even be possible for the current user to fetch DET (presumably by cloning) and end up with files not readable by the current user. I get the problem with execute permissions (which could either be absent in the repo, or get lost based on clone options, target FS, OS, and who knows what else). I've just never encountered a scenario where user read permissions were absent (in a normal file, that is).

Reply:
Updated to make more concise. I included the read permission in case it is needed in the future.